cpp/util/etcd.cc

```diff
@@ -180,6 +191,42 @@ void GetRequestDone(const string& keyname, EtcdClient::GetResponse* resp,
 }
 
 
+void CopyStat(const string& key, const JsonObject& from,
+              map<string, int64_t>* to) {
+  CHECK_NOTNULL(to);
+  const JsonInt stat(from, key.c_str());
+  if (!stat.Ok()) {
+    LOG(WARNING) << "Failed to find stat " << key;
+  }
+  (*to)[key] = stat.Value();
+}
+
+
+void GetStoreStatsRequestDone(EtcdClient::StatsResponse* resp,
+                              Task* parent_task,
+                              EtcdClient::GenericResponse* gen_resp,
+                              Task* task) {
+  *resp = EtcdClient::StatsResponse();
+  resp->etcd_index = gen_resp->etcd_index;
+  if (!task->status().ok()) {
+    parent_task->Return(Status(task->status().CanonicalCode(),
```

Comment:
If you're not modifying the status, you should just return it directly:

    parent_task->Return(task->status());